0000-deprecate-dependency-api.md

```diff
@@ -0,0 +1,40 @@
+- Feature Name: Deprecating the RubyGems.org dependency API
+- Start Date: 2023-01-27
+- RFC PR: (leave this empty)
+- Bundler Issue: (leave this empty)
+
+# Summary
+
+Plan for the deprecation and removal of the "dependency API" (`/api/v1/dependencies`) on rubygems.org.
+
+# Motivation
+
+The dependency API was the primary way Bundler fetched dependency info for Gemfile resolution until the release of [Bundler 1.12](https://bundler.io/blog/2016/04/28/the-new-index-format-fastly-and-bundler-1-12.html) in April 2016, which introduced the ["new" Compact Index API](https://andre.arko.net/2014/03/28/the-new-rubygems-index-format/).
+
+The Compact Index was designed to be completely cacheable and served at the edge by Fastly. In contrast, the dependency API is completely _uncacheable_, and requires RubyGems.org to run a complex database query for each request, accounting for almost 50% of the total uncached request volume to RubyGems.org with nearly 3000qps.
+
+Because the Compact Index has been out for over half a decade and Bundler & RubyGems both support fallback to the "full" index (`specs.4.8.gz` files) when the Dependency API is unavailable, we plan to deprecate and delete the dependency API over the coming months.
```

Comment:
🤔 Is there any known gem server (public or private) supporting Dependency API (in production graded state) as well?

Reply:
I think gemstash and artifactory both support the dependency API, but artifactory also (as of the latest version) supports the compact index

Reply:
I think it's also OK to leave in client support for the dependency API (rubygems or bundler or both) so that existing servers can continue to provide that benefit if they do not yet support the compact index. The main goal here is to remove support from rubygems.org so that we can remove a bunch of deprecated code paths. It will make Bundler versions released between 2011 and 2016 somewhat slower to install, but I think that is OK now. We should ask anyone who complains to upgrade to Bundler newer than 7 years old. :)

Reply:
Yes, I was about to make it suuuuuuuuuper clear it is not going to be removed from clients for now.